plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/ReadSessionCreator.java

```diff
@@ -102,14 +105,15 @@ public ReadSession create(ConnectorSession session, TableId remoteTable, List<Bi
             }
             // At least 100 to cater for cluster scale up
             int desiredParallelism = Math.min(currentWorkerCount * 3, 100);
-            CreateReadSessionRequest createReadSessionRequest = CreateReadSessionRequest.newBuilder()
+            CreateReadSessionRequest.Builder requestBuilder = CreateReadSessionRequest.newBuilder()
                     .setParent("projects/" + client.getParentProjectId())
                     .setReadSession(ReadSession.newBuilder()
                             .setDataFormat(format)
                             .setTable(toTableResourceName(actualTable.getTableId()))
-                            .setReadOptions(readOptions))
-                    .setPreferredMinStreamCount(desiredParallelism)
-                    .build();
+                            .setReadOptions(readOptions));
+            maxParallelism.ifPresentOrElse(
+                    requestBuilder::setMaxStreamCount,
+                    () -> requestBuilder.setPreferredMinStreamCount(desiredParallelism));
```

Comment:
Shouldn't we always set min stream count, making sure it's not greater than max stream count? Both of these numbers can be pretty high.

Ref: https://cloud.google.com/php/docs/reference/cloud-bigquery-storage/latest/V1.CreateReadSessionRequest#_Google_Cloud_BigQuery_Storage_V1_CreateReadSessionRequest__setMaxStreamCount__

Reply:
Good question. I was wondering the same thing. Since pretty high parallelism caused cluster blocking issue #25423, I expected max-stream config will be used with a low value for limiting preferred-min-stream until the issue is fixed. 

Anyway, I will push a change to always set preferred-min-stream. I know Spark BigQuery connector always sets it.